client/JSDebugger/mlRuntime.ts

```diff
@@ -245,70 +247,104 @@ export class MLRuntime extends EventEmitter {
     private _sendMLdebugRequestPOST(module: string, body?: string): Promise<string> {
-        const url = this.buildUrl(`/${module}/${this._rid}`);
-        const options: Record<string, unknown> = {
-            headers: {
-                'Content-type': 'application/x-www-form-urlencoded',
-                'X-Error-Accept': 'application/json'
-            },
-            auth: {
-                user: this._username,
-                pass: this._password,
-                'sendImmediately': false
-            }
-        };
-        if (body) {
-            options['body'] = body;
-        }
-        if (this._ca) options['agentOptions'] = { ca: this._ca };
-        options['rejectUnauthorized'] = this._rejectUnauthorized;
-        return request.post(url, options);
+        const manageClientParams: MlClientParameters = JSON.parse(JSON.stringify(this.dbClientContext.params));
+        manageClientParams.port = this.managePort;
+        manageClientParams.contentDb = null;
+        manageClientParams.modulesDb = null;
+        const manageClient = new ClientContext(manageClientParams);
+
+        return new Promise((resolve, reject) => {
+            manageClient.databaseClient.internal.sendRequest(
+                `/jsdbg/v1/${module}/${this._rid}`,
+                (requestOptions: ml.RequestOptions) => {
+                    requestOptions.method = 'POST';
+                    requestOptions.headers = { 'X-Error-Accept': 'application/json' };
+                },
+                (operation: ml.RequestOperation) => {
+                    operation.requestBody = body;
+                })
+                .result((result: object) => {
+                    let response = '';
+                    if (result) {
+                        response = JSON.stringify(result);
+                    }
+                    resolve(response);
+                }).catch(error => {
+                    reject(error.body.errorResponse);
+                });
+        });
     }
 
-    private _sendMLdebugRequestGET(module: string, queryString?: Record<string, unknown>): Promise<string> {
-        const url = this.buildUrl(`/${module}/${this._rid}`);
-        const options: Record<string, unknown> = {
-            headers: {
-                'X-Error-Accept': 'application/json'
-            },
-            auth: {
-                user: this._username,
-                pass: this._password,
-                'sendImmediately': false
-            }
-        };
-        if (queryString) {
-            options['qs'] = queryString;
+    private _sendMLdebugRequestGET(module: string, queryStringObject?: Record<string, unknown>): Promise<string> {
+        const manageClientParams: MlClientParameters = JSON.parse(JSON.stringify(this.dbClientContext.params));
+        manageClientParams.port = this.managePort;
+        manageClientParams.contentDb = null;
+        manageClientParams.modulesDb = null;
+        const manageClient = new ClientContext(manageClientParams);
```

Comment:
These 5 lines are repeated above - perhaps a `newManageClient` function is warranted here?